float32 x
float32 z
bool jump
---
bool state